Add xml:Text() and annotations change (#101)
ballerina-platform/ballerina-release: release-notes/swan-lake-alpha3-release-note.md

#### Language

> ##### Improvements in annotations support

Reviewer: ```suggestion
##### Improvement to Annotation Attachment with Empty Mapping Constructor Expression
```

> `mapping-constructor-expr` is now allowed be optional in annotations, if the annotation type requires a map.

Reviewer 2: ```suggestion
The `mapping-constructor-expr` is now allowed to be optional in annotations if the annotation type requires a map.
```

Reviewer 3: What we want to say is: annotation value is now allowed to be omitted from an annotation, if the annotation value is empty mapping constructor.
@MaryamZi  please help us word this.

Reviewer 2: Let's simply say that. :)

"Now, it is optional to have the `mapping-constructor-expr` in an annotation if the annotation value is an empty mapping constructor."

Reviewer: Shall we say something like

```suggestion
If the type of the annotation is a mapping type for which an empty mapping constructor is valid, the mapping constructor expression is no longer mandatory in the annotation attachment.

The absence of the mapping constructor expression in such an annotation attachment is equivalent to specifying a mapping constructor expression with no fields.
```

```ballerina
type Annot record {|
    int[] i = [];
|};

public annotation Annot v1 on function;

> @v1

Reviewer: ```suggestion
@v1 // Same as `@v1 {}`
```

> public function main(string... argv) {

Reviewer: ```suggestion
public function main() {
```

}
```

> 
> ##### New `xml:text()` function

Reviewer: I think we'll have to add this under a separate langlib section.

```suggestion

##### Improved lang library functions

###### New `xml:text()` function
```

> This method can be used to select all the items in a sequence that are of type `xml:Text`.

Reviewer: ```suggestion
This function can be used to select all the items in a sequence that are of type `xml:Text`.
```
